src/dkpy/uncertainty_structure.py

```diff
@@ -0,0 +1,200 @@
+"""Uncertainty block."""
+
+import numpy as np
+from typing import Sequence
+
+
+class UncertaintyBlock:
```

Comment:
IMO it's better to make this abstract and then override things like `is_diagonal` in the child classes. Essentially the subclasses are all just parent classes with different data at this point, which is not quite how I'd use inheritance

Think of it this way. If the only thing that matters is the different data, you should never need to check the type...

I think moving the block structure generation in here would make things clearer

Reply:
Oh that's a good point. Maybe? I'm not sure. I'd need to think about it more. But you can try it and see if it's an improvement or a mess